lib/document/Document.js

```diff
@@ -250,29 +275,12 @@ class Document {
   toJSON() {
     const rawDocument = this.toObject();
```

Comment:
I feel it's not cool that we are converting things two times. First in toObject and the second time in toJSON(). I would recommend to add an option to toObject like `encodedBuffers: true` to skip transformation and add EncodedBuffer#toJSON that will call toString internally so you don't need to transform them even here, due `JSON.stringify` will call it, in turn.